[BE-2911]: Added cache doc, and how to share files doc (#384)
appcircleio/appcircle-docusaurus: docs/integrations/common-workflow-steps/build-cache/cache-pull.md

---
title: Cache Pull 
description: Enhance your development process with Cache Pull to quickly retrieve and reuse stored data, boosting efficiency and performance.
tags: [cache pull, efficiency, dependencies, cache structure]
sidebar_position: 2
---

import Screenshot from '@site/src/components/Screenshot';

# Cache Pull

> [**Cache Push**](./cache-push) uploads the cache archive file to a remote location, as we explained in detail in the **Cache Push** step. On the other hand, **Cache Pull** downloads and extracts that archive file in the build pipeline, restoring all files and folders to their original locations.

Reviewer: Links should be the exact path.

Wrong:
`[**Cache Push**](./cache-push)`

Correct:
`[**Cache Push**](/workflows/common-workflow-steps/cache-push)`

:::warning

> [**Cache Push**](./cache-push) and **Cache Pull** components should work in coordination on the same cache file. Therefore, to download the pushed cache, the **Cache Pull** must have the same cache label as the **Cache Push**.

Reviewer: Links should be the exact path.

Wrong:
`(./cache-push)`

Correct:
`(/workflows/common-workflow-steps/cache-push)`

:::

:::info
If you need to use the cached folder in a different branch or a separate project, you have the capability to modify the values of `$AC_GIT_BRANCH` or `$AC_BUILD_PROFILE_ID`. For further information, please check out the following documentation:

> - [How to Share Files Between Pipelines](./how-to-share-file-between-pipelines)

Reviewer: Links should be the exact path.

Wrong:
`(./cache-push)`

Correct:
`(/workflows/common-workflow-steps/cache-push)`

> - [How to Share Files Between Build Profiles](./how-to-share-file-between-build-profiles)

Reviewer: Links should be the exact path.

Wrong:
`(./cache-push)`

Correct:
`(/workflows/common-workflow-steps/cache-push)`

> These variables can be adjusted within the [cache label](#input-variables) field, as indicated by the red highlight in the accompanying image. Simply replace them with the branch or project ID that corresponds to your intended usage.

Reviewer: Links should be the exact path.

Wrong:
`(./cache-push)`

Correct:
`(/workflows/common-workflow-steps/cache-push)`

---
title: Cache Pull 
metaTitle: Cache Pull
metaDescription: Cache Pull
sidebar_position: 2
---

import Screenshot from '@site/src/components/Screenshot';

# Cache Pull

> Cache push uploads the cache archive file to a remote location, as we explained in detail in the [**Cache Push**](https://docs.appcircle.io/workflows/common-workflow-steps/#cache-push) step. On the other hand, cache pull downloads and extracts that archive file in the build pipeline. All files and folders are extracted to the original locations they came from.

Reviewer 2: ```suggestion
Cache Push uploads the cache archive file to a remote location, as explained in detail in the [**Cache Push**](https://docs.appcircle.io/workflows/common-workflow-steps/#cache-push) step. On the other hand, Cache Pull downloads and extracts that archive file in the build pipeline, restoring all files and folders to their original locations.

:::warning

> Cache push and pull components should work in coordination on the same cache file. For this reason, in order to download the pushed cache, the cache pull must have the same cache label as the cache push.

Reviewer 2: ```suggestion
Cache push and pull components should work in coordination on the same cache file. Therefore, to download the pushed cache, the Cache Pull must have the same cache label as the Cache Push.
```

:::

:::info

> In the event that you need to utilize the cached folder in an alternate branch or a separate project, you have the capability to modify the values of `$AC_GIT_BRANCH` or `$AC_BUILD_PROFILE_ID`.

Reviewer 2: Please give reference for `how to share file between...` docs as a article?
```suggestion
If you need to use the cached folder in a different branch or a separate project, you have the capability to modify the values of $AC_GIT_BRANCH or $AC_BUILD_PROFILE_ID.
```

These variables can be adjusted within the **Cache Label** field, as indicated by the red highlight in the accompanying image. Simply replace them with the branch or project ID that corresponds to your intended usage.

<Screenshot url='https://cdn.appcircle.io/docs/assets/cache-01.png' />

:::

### Prerequisites

:::caution

> This component does not require any prerequisites step for operation. The only thing necessary for the component to work as expected is to use the cached files before the step in which they will be used. At the same time, another important point for this step to work is that the files to be used must have been cached in previous builds. 

Reviewer 2: ```suggestion
This component does not require any prerequisite steps for operation. The only thing necessary for the component to work as expected is to utilize the cached files before the step in which they will be used. Additionally, an important prerequisite for this step to function properly is that the files to be used must have been cached in previous builds.```

> For example in the screen shot, to use cached files for **Cocoapods**, **Cache Pull** step should be used before [**Cocoapods Install**](https://docs.appcircle.io/workflows/ios-specific-workflow-steps/cocoapods-install) step.

Reviewer 2: ```suggestion
For example, in the screenshot, to use cached files for Cocoapods, the Cache Pull step should be used before the [**Cocoapods Install**](https://docs.appcircle.io/workflows/ios-specific-workflow-steps/cocoapods-install) step.
```

<Screenshot url='https://cdn.appcircle.io/docs/assets/BE2911-pullOrder.png' />
:::

:::warning

> If there are no previously cached files and you want to use this step, the **Cache Pull** step will give a **`not found error`** because it cannot find the specified files at the remote location.

Reviewer 2: ```suggestion
If there are no previously cached files and you attempt to use this step, the **Cache Pull** step will result in a **`not found error`** because it cannot locate the specified files at the remote location.
```
